src/frontend/Dockerfile

```diff
@@ -12,43 +12,29 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# The Google App Engine python runtime is Debian Jessie with Python installed
-# and various os-level packages to allow installation of popular Python
-# libraries. The source is on github at:
-#   https://github.com/GoogleCloudPlatform/python-docker
-FROM gcr.io/google-appengine/python
+# Use the official Python docker container, slim version, running Debian
+FROM python:3.10-slim
+
+# Set our working directory
+WORKDIR /app
 
 # show python logs as they occur
 ENV PYTHONUNBUFFERED=0
 
-# Create a virtualenv for the application dependencies.
-RUN virtualenv -p python3.7 /env
-
-# Set virtualenv environment variables. This is equivalent to running
-# source /env/bin/activate. This ensures the application is executed within
-# the context of the virtualenv and will have access to its dependencies.
-ENV VIRTUAL_ENV /env
-ENV PATH /env/bin:$PATH
-
 # explicitly set a fallback log level in case no log level is defined by Kubernetes
 ENV LOG_LEVEL info
 
 # Install dependencies.
-ADD requirements.txt /app/requirements.txt
-RUN pip install -r /app/requirements.txt
+COPY requirements.txt .
+RUN pip install -r requirements.txt
 
 # This is a workaround to use an unreleased version of grpcio, so that the cloud trace exporter and the OpenTelemetry RequestsInstrumentor do not create an infinite request loop.
 RUN pip install --pre --upgrade --force-reinstall --extra-index-url \
```

Comment:
This line is from July 2020 (before I took on this project) and I'm truthfully not 100% sure what it does. Do you think it can be removed? Looks like it was to skirt around a buggy version of grpcio maybe?

Reply:
Probably yes. I'll iterate locally and update this PR

Reply:
Yup, this custom install maps to grpcio-1.31.0 which is very much no longer the most recent version.